libqpdf/QPDFObjectHandle.cc

```diff
@@ -1912,7 +1924,19 @@ QPDFObjectHandle::parseInternal(PointerHolder<InputSource> input,
 	    break;
 
 	  case QPDFTokenizer::tt_name:
-	    object = newName(token.getValue());
+	    {
+		std::string name = token.getValue();
+		object = newName(name);
+
+		if (name == "/Contents")
```

Comment:
I'm feeling like this is going to prevent decryption of all /Contents dictionary entries whose values are string, when I think we only want to prevent decryption of /Contents in signature dictionaries. I feel like maybe there is a better way to handle this case. I'm happy to discuss further. There is a lot of extra bookkeeping added to handle this case. It would be nice to have a test case with a map with a /Contents value of type string that is not in a signature dictionary to make sure it is handled properly. I'm not sure whether this would actually appear in a real file, but weird things can be found in real files, and it would be a shame for qpdf to be leaving some strings encrypted because it falsely assumes they are in signature dictionaries. That would be a serious bug because it would result in loss of data.

Reply:
I've added tests for non signature dictionary /Contents string (text string) in #377.

If /Contents is not in signature dictionary, the string is not excluded encryption and decryption.
The exclusion is only for /Contents string in signature dictionary.
https://github.com/qpdf/qpdf/blob/8ce344792b40422b3ac36dac658311179ca7beca/libqpdf/QPDFObjectHandle.cc#L2195